### PR TITLE
Primrose Installation Modification (Issue #17)

### PR DESCRIPTION
**Pull request by parul-l**
When attempting to install the `primrose` library, I received the following errors:

* `pg_config` missing error (related to installing `psycopg2`)
*  `libcgraph` missing (related to installing `pygraphviz`)

Since these packages are not required to run a primrose DAG, they were removed from the basic installation. Users now have the option to install primrose without requiring the additional dependencies. 
* `pip install primrose` installs primrose without postgres, plotting packages
* `pip install primrose[postgres, plotting]` installs the complete primrose package.

Questions/concerns:
* I'm not sure how to test this actually works (ie. how pypi testing works).
* In thinking about it now, I'm not sure if we would want to impose installing `postgres` or `libpq-dev`. Perhaps we do and so I didn't modify the `test_postgres_reader.py` (hence the failing tests)

### CODE REVIEW

**Comment by kzecchini** [comment]
@parul-l Looks like the way that psycopg2 import is handled is not working correctly. One of the postgres tests is still failing.

**Comment by briangrahamww** [comment]
> @parul-l Looks like the way that psycopg2 import is handled is not working correctly. One of the postgres tests is still failing.

Is this related to the `psycopg2` vs `psycopg2-binary` install issue?

**Comment by kzecchini** [comment]
@parul-l @briangrahamww 

Actually I think it is due to the fact that the requirements are taken out of `requirements.txt`. We need to still install these requirements when we bootstrap the travis environment for testing. They were moved to the the `extras_require` section of the `setup.py` file.

**Comment by parul-l** [comment]
Thanks for taking a look @kzecchini and @briangrahamww. Yes, I removed `psycopg2` and `psycopg2-binary` from the requirements since perhaps we would not want to impose users to download the appropriate dependencies in order for these installations to work. The intention was that you don't need postgres to run a `primrose` DAG. If it is better to just require the dependencies and only make plotting optional, that works too. I just adjusted the code given the issues I had when installing.

**Comment by parul-l** [comment]
> With this change for optional installs, we should probably have tests for having the module vs. not having the module.

^^ I am not sure how to test this. Any insight would be appreciated!

**Comment by kzecchini** [comment]
I think what you did is a nice addition @parul-l. For now maybe we can just modify the travis bootstrapping script to install the optional dependencies? 

For testing each individual install, I think maybe we can just have some additional travis setup for multiple environments - one for full install, one for postgres, and one for plotting. Then we can have some condition in pytest like a `skipif` which checks for the proper packages installed.

**Comment by parul-l** [comment]
Thanks @kzecchini . I did the following:
* In `test/test_configuration_dag.py` I added two markers on the test function that uses the `pygraphviz` function. One marker signifying it is part of the optional build, and the other signifying it is for the plotting build.
```
@pytest.mark.optional
@pytest.mark.plotting
def test_plot_dag():
   blah ...
```
* I did something similar in `test/test_postgres_reader.py` for `psycopg2`.
* In the `.travis.yml` file, I added four environment variables to be added to the matrix of our builds. These variables specify the type of pypi build we want (the whole package, only basic, only plotting, etc.). In the `install` section of the `.travis.yml`, I install all the python libraries needed for the basic install (`pip install -r requirements.txt`), and then add any additional libraries, depending on the pypi build variable. Finally, in the script section, I call the appropriate pytest command that will skip certain tests based on the pypi build we have. For example, `python -m pytest -m "not plotting"` will run all tests except for the tests marked "plotting".

File: .travis.yml [hunk 1]
```diff
@@ -27,8 +35,8 @@ install:
 # os dependencies and set up matplotlib backend
 # travis does not support python on osx - install via miniconda
 - |
-  if [ $TRAVIS_OS_NAME = 'osx' ]; then 
-    echo "Installing brew packages..."  
+  if [ $TRAVIS_OS_NAME = 'osx' ]; then
+    echo "Installing brew packages..."
     brew install graphviz
```
Comment (briangrahamww) [inline .travis.yml]: Shouldn't we only `brew install pygraphviz` `if [ $PYPI_BUILD = 'complete' ]` or `if [ $PYPI_BUILD = 'only_plotting' ]`

File: primrose/readers/postgres_helper.py [hunk 2]
```diff
@@ -1,15 +1,22 @@
 """Postgres helper
 
-Author(s): 
+Author(s):
     Wassym Kalouache (wassym.kalouache@weightwatchers.com)
 
     Carl Anderson (carl.anderson@weightwatchers.com)
```
Comment (briangrahamww) [inline primrose/readers/postgres_helper.py]: Don't forget to add yourself!

**Pull request merged**